ios/FluentUI/Notification/NotificationView_SwiftUI.swift

```diff
@@ -172,54 +159,45 @@ public struct NotificationViewSwiftUI: View {
     }
 
     public var body: some View {
-        let width = UIScreen.main.bounds.width - safeAreaInsets.leading - safeAreaInsets.trailing
-        innerContents
-            .onTapGesture {
-                if state.messageButtonAction != nil {
-                    activeAlert = .message
-                    showingAlert = true
-                }
-            }
-            .frame(width: state.style.isToast && horizontalSizeClass == .regular ? width / 2 : width - (2 * tokens.presentationOffset))
-            .background(
-                RoundedRectangle(cornerRadius: tokens.cornerRadius)
-                    .strokeBorder(Color(tokens.outlineColor), lineWidth: tokens.outlineWidth)
-                    .background(
-                        RoundedRectangle(cornerRadius: tokens.cornerRadius)
-                            .fill(Color(tokens.backgroundColor))
-                    )
-                    .shadow(color: Color(tokens.ambientShadowColor), radius: tokens.ambientShadowBlur, x: tokens.ambientShadowOffsetX, y: tokens.ambientShadowOffsetY)
-                    .shadow(color: Color(tokens.perimeterShadowColor), radius: tokens.perimeterShadowBlur, x: tokens.perimeterShadowOffsetX, y: tokens.perimeterShadowOffsetY)
-            )
-            .designTokens(tokens,
-                          from: theme,
-                          with: windowProvider)
-            .alert(isPresented: $showingAlert) {
-                var alertString: String
-                switch activeAlert {
-                case .actionButton:
-                    if let actionTitle = state.actionButtonTitle {
-                        if actionTitle.isEmpty {
-                            alertString = "Dismiss"
-                        } else {
-                            alertString = actionTitle
-                        }
-                    } else {
-                        alertString = "Action"
+        // Note: we are using viewController's width because GeometryReader relies on the bounds of the hosting controller which cannot be set accurately at this point
+        if let viewWidth = viewController?.view.bounds.width {
```

Comment:
One idea that came to my mind here. What if you use `windowProvider?.window?.bounds.width` here instead of passing the environment value just for that view controller?
Seems to me you'd be able to save a lot of code in the show method that presents it.